src/test/java/picard/sam/FilterSamReadsTest.java

```diff
@@ -56,22 +77,70 @@ public void testJavaScriptFilters(final String samFilename, final String javascr
         // input as SAM file 
         final File inputSam = new File(samFilename);
         final File javascriptFile = new File(javascriptFilename);
-        //loop over javascript filters
+
+        FilterSamReads filterTest = setupProgram(javascriptFile, inputSam, FilterSamReads.Filter.includeJavascript);
+        Assert.assertEquals(filterTest.doWork(),0);
+
+        int count = getReadCount(filterTest);
+
+        Assert.assertEquals(count, expectNumber);
+    }
+
+    /**
+     * filters a SAM using an interval filter
+     */
+    @Test(dataProvider = "dataTestPairedIntervalFilter")
+    public void testPairedIntervalFilter(final String intervalFilename, final int expectNumber) throws Exception {
+        // input as SAM file
+        final File inputSam = File.createTempFile("testSam", ".sam", TEST_DIR);
+        inputSam.deleteOnExit();
+
+        final SAMFileWriter writer = new SAMFileWriterFactory()
+                .setCreateIndex(true).makeBAMWriter(builder.getHeader(), false, inputSam);
+
+        for (final SAMRecord record : builder) {
+            writer.addAlignment(record);
+        }
+        writer.close();
+
+        final File intervalFile = new File(intervalFilename);
+
+        FilterSamReads filterTest = setupProgram(intervalFile, inputSam, FilterSamReads.Filter.includePairedIntervals);
+        Assert.assertEquals(filterTest.doWork(),0);
+
+        int count = getReadCount(filterTest);
+
+        Assert.assertEquals(count, expectNumber);
+    }
+
+    private FilterSamReads setupProgram(final File inputFile, final File inputSam, final FilterSamReads.Filter filter) throws Exception {
         final FilterSamReads program = new FilterSamReads();
         program.INPUT = inputSam;
         program.OUTPUT = File.createTempFile("FilterSamReads.output.", ".sam");
         program.OUTPUT.deleteOnExit();
-        program.FILTER = FilterSamReads.Filter.includeJavascript;
-        program.JAVASCRIPT_FILE = javascriptFile;
-        Assert.assertEquals(program.doWork(),0);
-        
-        //count reads
-        final SamReader samReader = SamReaderFactory.makeDefault().open(program.OUTPUT);
+        program.FILTER = filter;
+
+        if(filter == FilterSamReads.Filter.includePairedIntervals) {
+            program.INTERVAL_LIST = inputFile;
+        }
+        else {
+            program.JAVASCRIPT_FILE = inputFile;
+        }
+
+        return program;
+    }
+
+    private int getReadCount(FilterSamReads filterTest) throws Exception {
```

Comment:
I think this is a great function to java 8-ify (stream and count the iterator), if you want the practice